services/app-web/src/components/Form/FieldTextInput/FieldTextInput.tsx

```diff
@@ -16,35 +18,59 @@ import { PoliteErrorMessage } from '../../'
 
 export type TextInputProps = {
     label: string
+    variant?: 'TOPLEVEL' | 'SUBHEAD' // subhead used for forms where fields could be nested under a larger heading or label
     id: string
     hint?: React.ReactNode
     error?: string
     showError: boolean
     name: string
     type: 'text' | 'email' | 'number' | 'password' | 'search' | 'tel' | 'url'
-} & JSX.IntrinsicElements['input']
+} & React.ComponentProps<typeof TextInput>
 
 export const FieldTextInput = ({
+    className,
     label,
+    variant = 'TOPLEVEL',
     id,
     hint,
     error,
     showError,
     name,
     type,
+    onBlur,
     ...inputProps
 }: TextInputProps): React.ReactElement => {
-    const [field, meta] = useField({ name })
+    const [field, meta, helpers] = useField({ name })
+
+    const classes = classNames(
+        {
+            [styles.nestedField]: variant === 'SUBHEAD',
+        },
+        className
+    )
+
+    // Latch into onBlur to do any input cleaning
+    // Initial use case is to trim away trailing whitespace in email addresses
+    const customOnBlur = (e: React.FocusEvent<HTMLInputElement, Element>) => {
+        if (!e) return
+        helpers.setValue(field.value.trim())
```

Comment:
This `customOnBlur` does not seem to be working for email input on the contacts page. When not focused on the email input, it still retains leading and trailing spaces. 

You can continue to supporting documents with leading trailing spaces, but **it does trim**, even when `trim()` is removed from the the Yup validation in `Contacts.tsx`. I think is a graphical issue only.